test/serializer/optimized-functions/ConditionallyLeakedObject.js

```diff
@@ -0,0 +1,17 @@
+// copies of 42:1
+
+(function () {
+  function f(g, c) {
+    let o = {foo: 42};
+    if (c) {
+      g(o);
+    } else {
+      o.now = 1;
+    } 
+
+    return o;
+  }
+
+  global.__optimize && __optimize(f);
+  inspect = function() { return f(true, h => h()); }
```

Comment:
Just returning an object from the inspect function will not cause a deep inspection. You want to create a string which contains `.now` and `.foo`.